### PR TITLE
Blog - Auth with Loopback

### PR DESCRIPTION
**Pull request by icode247**
title: Implementing User Authentication and Authorization in a Loopback REST API

### CODE REVIEW

File: content/author.yaml [hunk 1]
```diff
@@ -584,3 +584,10 @@
   github: "Babatunde13"
   linkedin: "babatunde-koiki-2002"
   twitter: "bkoiki950"
+
+- id: Ekekenta Odionyenfe Clinton
+  bio: I am a software engineer and technical writer who is proficient in server-side scripting and database setup.
```
Comment (raghunath-r-a) [inline content/author.yaml:589]: Please try to expand your bio, ideally around 300 characters.

File: content/implementing-user-authentication-and-authorization-in-a-loopback-rest-api/index.md [hunk 2]
```diff
@@ -0,0 +1,542 @@
+---
+title: "Implementing User Authentication and Authorization in a Loopback REST API"
+date: "2021-12-17"
+coverImage: "coverImage.jpg"
```
Comment (raghunath-r-a) [inline content/implementing-user-authentication-and-authorization-in-a-loopback-rest-api/index.md:4]: Please change `coverImage.jpg` to something relevant to the topic of Loopback. For example: https://www.loginradius.com/blog/async/guest-post/securing-flask-api-with-jwt/

File: content/implementing-user-authentication-and-authorization-in-a-loopback-rest-api/index.md [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+- [**MongoDB**](https://www.googleadservices.com/pagead/aclk?sa=L&ai=DChcSEwiiqMOcgOT0AhW9E9QBHX_NBkwYABAAGgJvYQ&ae=2&ohost=www.google.com&cid=CAESQOD2-dSvIZjrPJ6ygmiUs_a0EdOfNl4UZ08K2P8CBdRlG8XM7z1URxeAPsMds5XnNZzxbl6wm8XmIxQ2YIdzJP8&sig=AOD64_1wDL4266dBtlF-1EjB6lfvSLjIvw&q&adurl&ved=2ahUKEwiNn7icgOT0AhW0nGoFHUPjDGgQ0Qx6BAgDEAE) - Document-oriented database program.
+
+
+## Install Loopback CLI
+
+To start building our Loopback REST API, we must first install the Loopback CLI, which provides the quickest method to get started with a LoopBack 4 project that follows best practices.
+Use the command below to install the Loopback CLI globally:
+
+
+    npm i -g @loopback/cli
```
Comment (raghunath-r-a) [inline content/implementing-user-authentication-and-authorization-in-a-loopback-rest-api/index.md:53]: Please use syntax highlighting with markdown code blocks.

**Review by raghunath-r-a — CHANGES_REQUESTED** [review]
Hi, @icode247

I have added some initial comments for you to consider. 

Also, please move your blog folder inside the `./content/guest-post/` directory.

Please review them and update accordingly for us to move forward.

Thanks,
@raghunath-r-a

**Comment by icode247** [comment]
Hello Raghunath.
I have committed the changes on my draft.

Thanks

File: content/guest-post/loopback-rest-api-authentication/index.md [hunk 4]
```diff
@@ -0,0 +1,577 @@
...
+
+Then, create a [Repository](https://loopback.io/doc/en/lb4/Repository.html) for CRUD operations of your NewModel with the command below:
+
+```bash
+  lb4 repository
+```
+
+After completing the prompts, LoopBack will create the `NewsModelRepository` file in the `src/repository` folder.
+
+Select `NewsDatasource` as the data source, `NewsModel` as the to generate repository for, and `DefaultCrudRepository` as the base repository class.
```
Comment (raghunath-r-a) [inline content/guest-post/loopback-rest-api-authentication/index.md:117]: @icode247 This one phrase is little bit unclear: 

```
`NewsModel` as the to generate repository for
```
`as the to` is ambiguous here

Comment (akaashsaxena2897) [reply content/guest-post/loopback-rest-api-authentication/index.md:117]: Facing following error after running command `npm start` which is run after `lb4 controller`  command.

```

> auth-with-loopback@0.0.1 prestart
> npm run rebuild


> auth-with-loopback@0.0.1 rebuild
> npm run clean && npm run build


> auth-with-loopback@0.0.1 clean
> lb-clean dist *.tsbuildinfo .eslintcache


> auth-with-loopback@0.0.1 build
> lb-tsc

src/repositories/news-model.repository.ts:7:3 - error TS2344: Type 'NewsModel' does not satisfy the constraint 'Entity'.
  Type 'NewsModel' is missing the following properties from type 'Entity': getId, getIdObject 

7   NewsModel,
    ~~~~~~~~~

src/repositories/news-model.repository.ts:14:11 - error TS2345: Argument of type 'typeof NewsModel' is not assignable to parameter of type 'typeof Entity & { prototype: NewsModel; }'.
  Type 'typeof NewsModel' is missing the following properties from type 'typeof Entity': getIdProperties, getIdOf, buildWhereForId

14     super(NewsModel, dataSource);
             ~~~~~~~~~
```

Please help understand and how to resolve this error

**Comment by icode247** [comment]
> Greetings, I think there's some confusion in image for `lb4 controller` step to create `Controller`. Here, we create a `id` with type `string`, which for some reason is interfering with other `id` component of type `number` that we created before Screenshot 2022-01-25 at 2 45 37 PM
> 
> The screen shot shows the error user faces when trying `npm run start` command after creating a controller. Could you kindly clarify this issue? @icode247 #

> @akaashsaxena2897 I had the wrong image on the **create model** section. I have fixed the issue, it should work now.

**Review by akaashsaxena2897 — APPROVED** [review]
The application seems to be working fine on technical side!

**Review by raghunath-r-a — APPROVED** [review]

**Pull request merged**